### DISCUSSION TITLE
[Jamulus] what to do with multiple mirrors as download

### DISCUSSION THREAD

**Discussion by ann0see** [Homebrew/discussions#2961] [Writing Formulae/Casks] [answered]
https://github.com/Homebrew/homebrew-cask/blob/master/Casks/jamulus.rb

Has two download mirrors (main: GitHub releases (https://github.com/jamulussoftware/jamulus/releases/tag/r3_8_1); second: SourceForge)
Moreover, there is a legacy version for old OS. Can/should this be integrated in the formula? How do other programs handle that?

**Comment by neutric** [comment, marked as answer]
> Has two download mirrors (main: GitHub releases (https://github.com/jamulussoftware/jamulus/releases/tag/r3_8_1); second: SourceForge)

There is no built-in way to specify mirrors afaik. If you are experiencing problems with SourceForge, switch the `url` over to GitHub or provide a link which does the mirror selection behind the scenes.

You'd have to manually specify a `livecheck` after switching over to GitHub as the default is matching based on tags and those on GitHub suggest a different `version` format. Since you are tagging pre-releases there, you'd need `strategy :github_latest` to filter those out.

> Moreover, there is a legacy version for old OS. Can/should this be integrated in the formula? How do other programs handle that?

In your specific case where a legacy version is released alongside the current version, use something like:
```
version "3.8.1"
if MacOS.version <= :sierra
  sha256 "..."
  url "..."
else
  sha256 "..."
  url "..."
end
```

To not have to repeat `url` (+`verified`), it might be nicer to leave those two where they are right now, that is out of the conditional, and do `url "https://downloads.sourceforge.net/llcon/jamulus_#{version}_mac#{artifact}.dmg",`. The `artifact` variable would then have to be set to `"_legacy"` in the first branch of the conditional and `""` in the second.

**Reply by ann0see** [reply]
The conditional sounds interesting. If I find time maybe I can open a PR. 
SourceForge is still a valid mirror but GH preferred. If it means too many changes I‘d leave it at SF.

**Reply by ann0see** [reply]
Ok: See https://github.com/Homebrew/homebrew-cask/pull/119262

**Reply by neutric** [reply]
Happy you got it working. Sorry for misguiding you slightly by suggesting "artifact" as variable name. I was so focused on finding a meaningful name for the variable that it didn't occur to me to check whether there might be any problems with it.

**Reply by ann0see** [reply]
No problem. Thanks for your help.
